django/forms/forms.py

```diff
@@ -387,15 +390,15 @@ def full_clean(self):
         self._post_clean()
 
     def _clean_fields(self):
-        for name, field in self.fields.items():
+        for name, bf in self._bound_items():
+            field = bf.field
             if field.disabled:
-                value = self.get_initial_for_field(field, name)
+                value = bf.initial
             else:
-                value = self._field_data_value(field, self.add_prefix(name))
+                value = bf.data
```

Comment:
This can be nicely flattened:
```python
            value = bf.initial if field.disabled else bf.data
```
Or even:
```python
            value = field.bound_data(bf.data, bf.initial)
```

I'm not sure about the latter as there are specialized versions for `FileField` and `JSONField`. I don't know whether that is desired here -- something that was previously overlooked -- or not. See my [comment on ticket-32923](https://code.djangoproject.com/ticket/32923#comment:1).

Reply:
Thanks for the suggestions, @pope1ni. Yes, I looked into the option of using `bound_data()` as a future change but don't think it's okay to use as it does nontrivial data processing. See e.g. `JSONField.bound_data()`: https://github.com/django/django/blob/37e8367c359cd115f109d82f99ff32be219f4928/django/forms/fields.py#L1251-L1259

Based on my look at the other code paths I got the impression the difference was needed / deliberate.